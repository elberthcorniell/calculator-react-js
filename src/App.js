import { Component } from 'react';
import { ButtonPanel, Display } from './components';
import { calculate } from './logic';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const newState = calculate(this.state, buttonName);
    this.setState(newState);
  }

  render() {
    const { total } = this.state;
    return (
      <>
        <Display result={total} />
        <ButtonPanel clickHandler={this.handleClick} />
      </>
    );
  }
}

export default App;
