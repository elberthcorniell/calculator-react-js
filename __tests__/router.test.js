import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from '../src/App';

describe('test', () => {
  test('Render as spected', () => {
    render(<App />);
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  test('Loads and displays home button', () => {
    render(<App />);
    expect(screen.getAllByText('Home')).toBeDefined();
  });

  test('Loads and displays Home h1', () => {
    render(<App />);
    fireEvent.click(screen.getAllByText('Home')[0]);
    expect(screen.getAllByRole('heading')[1]).toHaveTextContent('Home');
  });

  test('Loads and displays Quote button', () => {
    render(<App />);
    expect(screen.getAllByText('Quote')).toBeDefined();
  });

  test('Loads and displays Quote h1', () => {
    render(<App />);
    fireEvent.click(screen.getAllByText('Quote')[0]);
    expect(screen.getAllByRole('heading')[1]).toHaveTextContent('Quote');
  });

  test('Does not load previews h1', () => {
    render(<App />);
    expect(screen.getAllByRole('heading')[1]).not.toHaveTextContent('Home');
  });

  test('Loads and displays Calculator button', () => {
    render(<App />);
    expect(screen.getAllByText('Calculator')).toBeDefined();
  });

  test('Loads and displays calculator', () => {
    render(<App />);
    fireEvent.click(screen.getAllByText('Calculator')[0]);
    expect(screen.getAllByText('=')).toBeDefined();
  });

  test('Does not load previews h1', () => {
    render(<App />);
    expect(screen.getAllByRole('heading')[1]).not.toBeDefined();
  });
});
