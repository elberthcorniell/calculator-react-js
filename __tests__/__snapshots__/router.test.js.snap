// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Render as spected 1`] = `"<div><nav><h1>Math magicians</h1><ul><li><a href=\\"/\\">Home</a></li><li><a href=\\"/calculator\\">Calculator</a></li><li><a href=\\"/quote\\">Quote</a></li></ul></nav><h1>Home</h1><p>Lorem, ipsum dolor sit amet consectetur adipisicing elit. Dolores illo error excepturi velit vero sunt ex, voluptate placeat qui pariatur minima quidem nemo est quia fugit sed laborum recusandae esse similique nostrum assumenda. Deserunt distinctio quod, delectus aspernatur voluptas rerum. Nesciunt unde maiores iste expedita magni non necessitatibus consectetur accusantium, sunt iure beatae incidunt obcaecati. Non cumque iure voluptatem iste id? Nam aperiam hic alias doloribus est. Omnis, cupiditate temporibus?</p></div>"`;
